e2etest/bgp_container.go

```diff
@@ -0,0 +1,189 @@
+	BGPDConfigFileName = "bgpd.conf"
+	// FRRRoutingImage contains the name and tag of the FRR routing image.
+	FRRRoutingImage = "frrouting/frr:v7.5.1"
+	// FRRContainerMountDestination contains the mount destination path.
+	FRRContainerMountDestination = "/etc/frr"
+)
+
+// Run a BGP router in a container.
+// Returns the IPv4 and IPv6 addresses of the container.
+func startFRRContainer(cs clientset.Interface, containerName string, testDirName string, bgpConfig string) (string, string, error) {
+	srcFiles := fmt.Sprintf("%s/.", BGPTestFRRConfigDir)
+	res, err := exec.Command("cp", "-r", srcFiles, testDirName).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to copy FRR config directory. %s", string(res))
+	}
+
+	err = setBGPConfig(testDirName, bgpConfig)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to update BGP config file")
+	}
+
+	volume := fmt.Sprintf("%s:%s", testDirName, FRRContainerMountDestination)
+	out, err := exec.Command("docker", "run", "-d", "--privileged", "--network", "kind", "--rm", "--name", containerName,
+		"--volume", volume, FRRRoutingImage).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to start %s container. %s", containerName, out)
+	}
+
+	containerIP, err := exec.Command("docker", "inspect", "-f", "{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}",
+		containerName).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to get FRR IP address")
+	}
+
+	containerIPv4 := strings.TrimSuffix(string(containerIP), "\n")
+
+	containerIP, err = exec.Command("docker", "inspect", "-f", "{{range .NetworkSettings.Networks}}{{.GlobalIPv6Address}}{{end}}",
+		containerName).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to get FRR IP address")
+	}
+
+	containerIPv6 := strings.TrimSuffix(string(containerIP), "\n")
+
+	return containerIPv4, containerIPv6, nil
+}
+
+// Change volume permissions.
+// Allows deleting the test directory or updating the files in the volume.
+func updateFRRContainerVolumePermissions(containerName string) error {
+	out, err := exec.Command("id", "-u").CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get UID")
+	}
+	uid := strings.TrimSuffix(string(out), "\n")
+
+	out, err = exec.Command("id", "-g").CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get GID")
+	}
+	gid := strings.TrimSuffix(string(out), "\n")
+
+	cmd := fmt.Sprintf("chown -R %s:%s %s", uid, gid, FRRContainerMountDestination)
+	out, err = exec.Command("docker", "exec", containerName, "sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to change %s container volume permissions. %s", containerName, string(out))
+	}
+
+	return nil
+}
+
+// Delete the BGP router configuration.
+func stopFRRContainer(containerName string, testDirName string) error {
+	err := updateFRRContainerVolumePermissions(containerName)
```

Comment:
Calling this right after the creation would make it possible to edit the file as part of the tests, right?